firmware.CPP

```diff
@@ -23,225 +25,220 @@ Adafruit_NeoPixel pixels(NUMPIXELS, PIN, NEO_GRB + NEO_KHZ800);
-  }
-  if (sound_on == true){
-     tone(5,1000);
-  }
-  delay(DASHLEN);
-  digitalWrite(LEDpin, LOW) ;
-  if (light_on == true){
-    beacon_off();
-  }
-   if (sound_on == true){
-    noTone(5);
-  }
-  delay(DOTLEN);
+void dash() {
+    checkMessage();
+    digitalWrite(LED_PIN, HIGH);
+    if (light_on) {
+        beacon_on();
+    }
+    if (sound_on) {
+        tone(BUZZER_PIN, 1000);
+    }
+    delay(DASHLEN);
+    digitalWrite(LED_PIN, LOW);
+    if (light_on) {
+        beacon_off();
+    }
+    if (sound_on) {
+        noTone(BUZZER_PIN);
+    }
+    delay(DOTLEN);
 }
 
-void
-dit()
-{
-  checkmessage();
-  digitalWrite(LEDpin, HIGH) ;
-  if (light_on == true){
-     beacon_on();
-   }
-  if (sound_on == true){
-     tone(5,1000);
-  }
-  delay(DOTLEN);
-  digitalWrite(LEDpin, LOW) ;
-  if (light_on == true){
-    beacon_off();
-  }
-  if (sound_on == true){
-    noTone(5);
-  }
-  delay(DOTLEN);
+void dit() {
+    checkMessage();
+    digitalWrite(LED_PIN, HIGH);
+    if (light_on) {
+        beacon_on();
+    }
+    if (sound_on) {
+        tone(BUZZER_PIN, 1000);
+    }
+    delay(DOTLEN);
+    digitalWrite(LED_PIN, LOW);
+    if (light_on) {
+        beacon_off();
+    }
+    if (sound_on) {
+        noTone(BUZZER_PIN);
+    }
+    delay(DOTLEN);
 }
 
-void
-send(char c)
-{
-  wdt_reset();  // reset the watchdog
-  int i;
-  if (c == ' ') {
-    Serial.print(c);
-    delay(7*DOTLEN);
-    return;
-  }
-  for (i=0; i<N_MORSE; i++) {
-    if (morsetab[i].c == c) {
-      unsigned char p = morsetab[i].pat ;
-      Serial.print(morsetab[i].c) ;
-
-      while (p != 1) {
-          if (p & 1)
-            dash() ;
-          else
-            dit() ;
-          p = p / 2 ;
-      }
-      delay(2*DOTLEN) ;
-      return ;
-    }
-  }
-  /* if we drop off the end, then we send a space */
-  Serial.print("?") ;
+void send(char c) {
+    wdt_reset();  // reset the watchdog
+    int i;
+    if (c == ' ') {
+        Serial.print(c);
+        delay(7 * DOTLEN);
+        return;
+    }
+    for (i = 0; i < N_MORSE; i++) {
+        if (morseTab[i].c == c) {
+            unsigned char p = morseTab[i].pat;
+            Serial.print(morseTab[i].c);
+
+            while (p != 1) {
+                if (p & 1)
+                    dash();
+                else
+                    dit();
+                p = p / 2;
+            }
+            delay(2 * DOTLEN);
+            return;
+        }
+    }
+    /* if we drop off the end, then we send a space */
+    Serial.print("?");
 
 }
 
-void
-sendmsg(char *str)
-{
-  while (*str)
-    send(*str++) ;
-  Serial.println("");
+void sendMsg(char *str) {
+    while (*str)
+        send(*str++);
+    Serial.println("");
 }
 
 
-//Radio shtuff
-RH_ASK driver(2000, 7, 12, 5); //Which will initialise the driver at 2000 bps, recieve on GPIO7, transmit on GPIO12, PTT on GPIO5. This is very poorly documented. PTT is not needed, but whatever. 
+//Radio stuff
+RH_ASK driver(2000, 7, LED_PIN, BUZZER_PIN);
```

Comment:
@TheBoss313 So the receive command pin is unused really

Reply:
@TheBoss313 Sorry for my poor documentation. The final argument (PTT) probably should be an unused pin (i.e 4), not the buzzer pin.

Reply:
I'm juggling between a custom modulation (CW, the morse section) and ASK packets via the RadioHead library

Reply:
Probably I'll switch to something else soon. Leave this for now though, could you please change the PTT pin though>

Reply:
Sure, will do

Reply:
What I meant was something like RH_ASK driver(2000, 7, 12, 1); perhaps that might work better...

Reply:
Using pins 0 and 1 on Arduino is a bad idea if you plan to use the USB port, as the communication happens to be on those pins